trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/screen/user_reel/UserReelViewModel.kt

```diff
@@ -129,6 +130,34 @@ internal class UserReelViewModel(
         )
     }
 
+    override fun saveUserReelEngagement(
+        reelWatchSessionState: ReelWatchSessionState,
+        reelId: String
+    ) {
+        tryToExecute(
+            block = {
+                reelsRepository.saveUserEngagementWithReel(
+                    prepareReelData(
+                        reelWatchSessionState,
+                        reelId
+                    )
+                )
+            },
+            dispatcher = defaultDispatcher
+        )
+    }
+
+    private fun prepareReelData(
+        reelWatchSessionState: ReelWatchSessionState,
+        reelId: String
+    ): ReelWatchSession {
+        val percentageOfVideoWatched =
+            reelWatchSessionState.watchedDurationInMilliseconds.toFloat() / reelWatchSessionState.videoDurationInMilliseconds * 100
+        reelWatchSessionState.reelId = reelId
+        return reelWatchSessionState.toEntity(percentageOfVideoWatched)
+    }
+
+    private fun onGetRefreshVideoUrl(refreshedUrl: String, reelId: String) {
```

Comment:
There is an issue here, probably stemmed from merging or resolving a conflict, kindly check and re-request the review.